pcdsdevices/lodcm.py

```diff
@@ -187,3 +403,389 @@ def remove_dia(self, moved_cb=None, timeout=None, wait=False):
         return status
 
     remove_dia.__doc__ += insert_remove
+
+    @pseudo_position_argument
+    def forward(self, pseudo_pos):
+        energy = self.get_energy()
+        # TODO this seems ridiculous, needs revising
+        return self.RealPosition(th1=energy, h1n=None, yag=None,
```

Comment:
I am not sure how the mapping will work here

Reply:
We need to re-frame this, this sort of construct is going to be rough

Only the `PositionerBase` subclasses will be picked up in the `RealPosition` tuple, which will cut down on this quite a bit... but it's still a bit ridiculous

Maybe we can find an alternate way of  implementing the offset motors to bring these calculation areas down to just handling the energy psuedomotor?

Reply:
Oh shoot I'm totally off-base, the state positioners ARE `PositionerBase`! That's awful

Reply:
Honestly this sounds like a terrible experience
Maybe we need to make a subdevice `PseudoPositioner` for `energy` instead of defining the whole LODCM as a `PseudoPositioner`, and just duplicate the motor components we need for the calculations

Reply:
Then instead of using `PseudoSingle` for the offset motors we have some sort of `OffsetIMS` class

Reply:
It looks like we really need to fix up ophyd's support for limiting the real positioners used in calculations. The alternative you propose sounds very reasonable, though.

Reply:
I am running into circles here with the LODCMEnergy class:
-  if I add the Offset motors after the `super().__init...` then it will not recognize them as real motors when `return self.RealPosition(...)`
 - if I define them as real components before at `init` then  `inverse` and `forward` will complain that I am missing some real motors, because I don't set all the motors, only set half of them depending of `Si` or `C`

Reply:
I only deal with `Offset Motors` when setting/getting energy..